server/src/main/java/io/crate/statistics/ReservoirSampler.java

```diff
@@ -89,20 +93,59 @@ public final class ReservoirSampler {
     private CircuitBreakerService circuitBreakerService;
     private final IndicesService indicesService;
 
+    private volatile ByteSizeValue readLimit;
+    private volatile ByteSizeValue updatedReadLimit;
+    private volatile boolean analyzing;
+
+    private final RateLimiter rateLimiter;
+    private long bytesSinceLastRateLimit = 0;
+
     @Inject
     public ReservoirSampler(ClusterService clusterService,
                             NodeContext nodeCtx,
                             Schemas schemas,
                             CircuitBreakerService circuitBreakerService,
-                            IndicesService indicesService) {
+                            IndicesService indicesService,
+                            Settings settings) {
         this.clusterService = clusterService;
         this.nodeCtx = nodeCtx;
         this.schemas = schemas;
         this.circuitBreakerService = circuitBreakerService;
         this.indicesService = indicesService;
+
+        this.readLimit = STATS_SERVICE_THROTTLING_SETTING.get(settings);
+        this.rateLimiter = new RateLimiter.SimpleRateLimiter(readLimit.getMbFrac());
+        this.updatedReadLimit = readLimit;
+        this.analyzing = false;
+
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(
+            STATS_SERVICE_THROTTLING_SETTING, this::updateReadLimit);
+    }
+
+    private void updateReadLimit(ByteSizeValue newReadLimit) {
+        boolean isAnalyzing = analyzing; // One volatile read.
+        if (isAnalyzing == false) {
+            readLimit = newReadLimit;
+            rateLimiter.setMBPerSec(readLimit.getMbFrac());
+        } else {
+            updatedReadLimit = newReadLimit; // One volatile write. Let running ANALYZE operation finish with the previous value.
```

Comment:
Yes I see no reason to not update a running analyze, I think this is even wanted like adjust it as the current analyze is slow and there is currently enough headroom to run it faster.

Reply:
I think it would actually be acceptable to _not_ update the throttling value of a running analyze, but no strong opinion.

But it would have the advantage that we could instantiate the RateLimiter on a new call with the current value from the cluster setting. No need for an extra setting-update-consumer or keeping track of multiple volatile variables.

Reply:
I let to update on the fly in https://github.com/crate/crate/pull/13046/commits/ebd3d49754ebdf208efc5c59a8fb801e692d5c31

motivation is to let users adjust the new setting optimally for their environment/setup. Would also mention in docs smth like 

 _"setting can be adjusted optimally for the concrete setup by running ANALYZE and playing around with the setting and monitoring cluster state at the same time"_
 
 currently there is no volatile variables other than `mbPerSec` which is used internally in `SimpleRateLimiter`